src/servers/src/grpc.rs

```diff
@@ -64,13 +68,15 @@ pub struct GrpcServerConfig {
     pub max_recv_message_size: usize,
     // Max gRPC sending(encoding) message size
     pub max_send_message_size: usize,
+    pub tls: TlsOption,
```

Comment:
Why `serve_state` and `handlers` require `Mutex`, and `tls_config` also require `Mutex`?

Reply:
The comment says it's mutable.

Reply:
for `tls_config` we have a feature to watch tls certficates and keys for changes and do live reloading without restart the server. I think for grpc we need to keep this behaviour.

Reply:
> we have a feature to watch tls certficates and keys for changes and do live reloading without restart the server

@sunng87 Hey, I'm a bit confused. Currently, `Frontend` sets `grpc TLS configuration` when the service starts up.
https://github.com/GreptimeTeam/greptimedb/blob/7da92eb9eba63ce3f5757b86fdd151a2602cced0/src/servers/src/grpc.rs#L175-L179
 However, if we observe changes to the config after the service has started, it seems there's no way to reload the configuration without restarting the grpc service?

Reply:
I haven't look into how to configure TLS for tonic. But for mysql/postgres servers, we configure it using `ReloadableTlsServerConfig` (you can search for its definition and usage). 

For a TLS server, you don't have to restart the server to make the new certificates/keys live. It's mostly because how API designed. Actually the tls configuration for server is a stateless thing. We just need to make sure we use latest version of configuration when accepting new connection.

Reply:
By checking the builder pattern of tonic api, I think it will be difficult to get the auto reload feature work for tonic based gRPC. 

If we cannot have that reload feature supported, we will need to raise an error when user configured `watch: true` in `TlsOption` for gRPC.